### PR TITLE
docs(apisix): wrote configure-routes

### PR DESCRIPTION
**Pull request by An-DJ**

### CODE REVIEW

**Review by juzhiyuan — COMMENTED** [review]
先 Review 了这一部分

File: apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md [hunk 1]
```diff
@@ -3,4 +3,93 @@ title: Configure Routes
...
+* Create a **Route** connecting to an example **Upstream**, which points to [httpbin](https://httpbin.org).
+* Use curl to echo a test request, showing how the APISIX proxies and forwards the API requests.
+
+## What's a Route
+
+A **Route** is a routing path to the upstream targets. In Apache APISIX, Route matchs the client's request based on defined rules, loads and executes the corresponding plugins, and forwards the request to the specified Upstream.
+
+A basic Route should have its id, uri and the corresponding Upstream.
+
+See [Route](#) for a description of how to configure the more detailed rules (such as methods, host and so on) and rich plugins.
```
[the remark this answers is not in the document]
Comment (An-DJ) [reply apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md:23]: 对我这里先预留了锚点，主要是这些 Concept 暂时还没有

File: apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md [hunk 2]
```diff
@@ -3,4 +3,93 @@ title: Configure Routes
...
+If you haven’t completed the [Get APISIX](./#get-apisix) step already, complete that before proceeding.
+
+## Create a Route
+
+The **Admin API** provides resources management capabilities such as Route and Upstream. In this section, we use Admin API to create a Route for forwarding to the [httpbin](https://httpbin.org), which rovides testing facilities for HTTP requests and responses.
+
+The Route with id `1` can be created by the command below. It matches all HTTP requests to `/anything/*` and forwards them to the Upstream of `httpbin`. The `X-API-KEY` is the authentication token of Admin API.
+
+```sh
+curl "http://127.0.0.1:9180/apisix/admin/routes/1" -H "X-API-KEY: edd1c9f034335f136f87ad84b625c8f1" -X PUT -d '
```
Comment (juzhiyuan) [inline apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md:45]: Use environment variable to disable API Key by default in quickstart shell.

File: apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md [hunk 3]
```diff
@@ -3,4 +3,93 @@ title: Configure Routes
...
+## Prerequisites
+This chapter is part of the Getting Started. It is recommended that you have followed the previous steps.
+
+If you haven’t completed the [Get APISIX](./#get-apisix) step already, complete that before proceeding.
+
+## Create a Route
+
+The **Admin API** provides resources management capabilities such as Route and Upstream. In this section, we use Admin API to create a Route for forwarding to the [httpbin](https://httpbin.org), which rovides testing facilities for HTTP requests and responses.
+
+The Route with id `1` can be created by the command below. It matches all HTTP requests to `/anything/*` and forwards them to the Upstream of `httpbin`. The `X-API-KEY` is the authentication token of Admin API.
```
Comment (juzhiyuan) [inline apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md:42]: anything -> http://httpbin.org/headers

Comment (moonming) [inline apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md:42]: We can add a picture here to explain the example of this route

Comment (moonming) [inline apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md:42]: `1` is not a good name for route.

File: apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md [hunk 4]
```diff
@@ -3,4 +3,93 @@ title: Configure Routes
...
+    }
+  }
+}'
+```
+
+Besides, Upstream can be defined separately from Route. See [Admin API](#) for a description of how to manage the resources like Route and Upstream.
+
+## Validation
+
+We can validate the Route created by curl:
```
Comment (moonming) [inline apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md:61]:
```suggestion
We can validate this route by curl:
```

File: apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md [hunk 5]
```diff
@@ -3,4 +3,93 @@ title: Configure Routes
...
+  "upstream": {
+    "type": "roundrobin",
+    "nodes": {
+      "httpbin.org:80": 1
+    }
+  }
+}'
+```
+
+Besides, Upstream can be defined separately from Route. See [Admin API](#) for a description of how to manage the resources like Route and Upstream.
```
Comment (moonming) [inline apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md:57]: remove this first. you can add it back when admin API exists.

File: apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md (hunk 4 above)
Comment (moonming) [inline apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md:61]: please change all `we` to `you`

File: apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md [hunk 6]
```diff
@@ -3,4 +3,93 @@ title: Configure Routes
...
+See [Upstream](#) for a description of detailed configuration such as load balancing, health check and retry policy.
+
+## Prerequisites
+This chapter is part of the Getting Started. It is recommended that you have followed the previous steps.
+
+If you haven’t completed the [Get APISIX](./#get-apisix) step already, complete that before proceeding.
+
+## Create a Route
+
+The **Admin API** provides resources management capabilities such as Route and Upstream. In this section, we use Admin API to create a Route for forwarding to the [httpbin](https://httpbin.org), which rovides testing facilities for HTTP requests and responses.
```
Comment (moonming) [inline apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md:40]:
```suggestion
 In this section, you use API to create a route for forwarding to the [httpbin](https://httpbin.org), which provides testing facilities for HTTP requests and responses.
```

**Review by moonming — CHANGES_REQUESTED** [review]

File: apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md [hunk 7]
```diff
@@ -3,4 +3,93 @@ title: Configure Routes
 slug: /getting-started/configure-routes
 ---
 
-Coming soon.
\ No newline at end of file
+Apache APISIX provides flexible gateway management capabilities based on the concept of route. The simplest gateway model is the **Route-Upstream** model. **Route** and **Upstream** define the routing paths and targets of the requests and responses, respectively.
```
Comment (moonming) [inline apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md:6]:
```suggestion
Apache APISIX provides flexible gateway management capabilities based on the concept of the route. Route and upstream define the routing paths and targets of the requests and responses, respectively.
```

File: apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md [hunk 8]
```diff
@@ -3,4 +3,93 @@ title: Configure Routes
...
+
+The high level overview below shows requests arriving at Route and being forward to Upstream, then responses take the opposite path back to the client.
+
+![Overview](https://static.apiseven.com/uploads/2023/02/03/LQ3FQYNb_route-upstream.jpg)
+
+This tutorial walks through how to create Route by **Admin API** and test it. You will complete the following steps:
+
+* Create a **Route** connecting to an example **Upstream**, which points to [httpbin](https://httpbin.org).
+* Use curl to echo a test request, showing how the APISIX proxies and forwards the API requests.
```
Comment (moonming) [inline apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md:15]:
```suggestion
* Create a route connecting to an example upstream, which points to [httpbin](https://httpbin.org).
* Use curl to echo a test request, showing how the APISIX proxies and forwards the API requests.
```

File: apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md [hunk 9]
```diff
@@ -3,4 +3,93 @@ title: Configure Routes
...
+
+## What's a Route
+
+A **Route** is a routing path to the upstream targets. In Apache APISIX, Route matchs the client's request based on defined rules, loads and executes the corresponding plugins, and forwards the request to the specified Upstream.
```
Comment (moonming) [inline apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md:19]:
```suggestion
A route is a routing path to the upstream targets. In Apache APISIX, the route matches the client's request based on defined rules, loads and executes the corresponding plugins, and forwards the request to the specified Upstream.
```

**Review by moonming — CHANGES_REQUESTED** [review]

File: apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md [hunk 10]
```diff
@@ -3,4 +3,93 @@ title: Configure Routes
...
+An **Upstream** is a set of target nodes with the same work. Upstream define the virtual host abstraction that performs load balancing on a given set of service nodes according to the configured rules.
+
+A basic Upstream should have its id and the corresponding nodes.
+
+See [Upstream](#) for a description of detailed configuration such as load balancing, health check and retry policy.
+
+## Prerequisites
+This chapter is part of the Getting Started. It is recommended that you have followed the previous steps.
+
+If you haven’t completed the [Get APISIX](./#get-apisix) step already, complete that before proceeding.
```
Comment (moonming) [inline apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md:36]: I don't think we need `Prerequisites`

**Review by moonming — DISMISSED** [review]

**Comment by juzhiyuan** [comment]
image

Let's use _Italic_ for new terms :)

**Review by juzhiyuan — APPROVED** [review]

**Comment by An-DJ** [comment]
CC @juzhiyuan A caution has been added to remind the user.

image

Besides, `[//]: <balabala>` maybe a useful tool to record the TODO item, such as the unsettled target link.

**Comment by An-DJ** [comment]
CC @moonming

**Comment by juzhiyuan** [comment]
Preview page: https://api7-docs-21.preview.api7.ai/apisix/getting-started/configure-routes

File: apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md [hunk 11]
```diff
@@ -3,4 +3,70 @@ title: Configure Routes
...
+
+Complete the [Get APISIX](./) step before proceeding.
+
+:::
+
+## What's a Route
+
+A Route is a routing path to the upstream targets. In [Apache APISIX](https://api7.ai/apisix), the Route matches the client's requests based on defined rules, loads and executes the corresponding plugins, then forwards the request to the specified upstream targets.
+
+One Route should have _uri_ and the corresponding Upstream at least.
```
Comment (shreemaan-abhishek) [inline apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md:23]:
```suggestion
One Route should have a _URI_ and the corresponding Upstream at least.
```

File: apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md [hunk 12]
```diff
@@ -3,4 +3,70 @@ title: Configure Routes
...
+Apache APISIX provides flexible gateway management capabilities based on _Route_, which defines the request's routing path and targets.
+
+This tutorial guides you on how to create one Route and validate it. You will complete the following steps:
+
+1. Create a Route with an example _Upstream_, which points to [httpbin.org](http://httpbin.org).
+2. Use _cURL_ to send a test request, showing how APISIX proxies and forwards the request.
+
+:::info
+
+Complete the [Get APISIX](./) step before proceeding.
```
Comment (shreemaan-abhishek) [inline apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md:15]: I am not sure if `./` would work. Please confirm.

File: apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md [hunk 13]
```diff
@@ -3,4 +3,70 @@ title: Configure Routes
...
+
+In this section, let's create a Route to forward client requests to the [httpbin.org](http://httpbin.org), which is a public HTTP Request and Response service.
+
+The following command creates one Route which matches all requests to the `http://127.0.0.1:9080/ip` endpoint, forwards them to the [httpbin.org/ip](http://httpbin.org/ip), and returns the requester's IP address.
+
+[//]: <TODO: Add the link to the authorization of Admin API>
+
+:::caution
+
+The authorization of API is switched off for a better experience in this tutorial. If you deploy the APISIX in the production environment, it's recommended to open it.
```
Comment (shreemaan-abhishek) [inline apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md:39]:
```suggestion
The authorization of API is switched off for a better experience in this tutorial. If you deploy the APISIX in the production environment, it's recommended to turn it on.
```

**Review by shreemaan-abhishek — CHANGES_REQUESTED** [review]

File: apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md (hunk 11 above)
Comment (juzhiyuan) [reply apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md:23]: Please keep `uri` as it's a parameter :)

File: apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md (hunk 12 above)
Comment (juzhiyuan) [reply apisix_versioned_docs/version-3.1.0/getting-started/configure-routes.md:15]: Yes, it works.

**Pull request merged**